### PR TITLE
Revert "Delete deprecated AutoML operators and deprecate AutoML hook and links (#38418)"

### PR DESCRIPTION
**Pull request by shahar1**
This reverts commit 4a4eee11 (#38418).

As @molcay stated in #38418, the operators and hook may still be useful for AutoML Translation, which I overlooked when working on the PR above. 
I'll create another PR to delete specific ops. and hooks that become unusable and to convert some deprecation warnings to exceptions.
I'll try to be more careful on similar occasions.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by eladkal** [comment]
I assume this is due to https://github.com/apache/airflow/pull/38418#issuecomment-2025027673 ?

Can you please clarify what APIs are not going to be useful after Mar 31? Because if the APIs are not useful we don't need to revert

**Comment by shahar1** [comment]
> I assume this is due to [#38418 (comment)](https://github.com/apache/airflow/pull/38418#issuecomment-2025027673) ?
> 
> Can you please clarify what APIs are not going to be useful after Mar 31? Because if the APIs are not useful we don't need to revert



> I assume this is due to [#38418 (comment)](https://github.com/apache/airflow/pull/38418#issuecomment-2025027673) ?
> 
> Can you please clarify what APIs are not going to be useful after Mar 31? Because if the APIs are not useful we don't need to revert

Yes, the following APIs won't be available: AutoML Tables, AutoML Video Intelligence, AutoML Vision and AutoML NLP.
However, there is one API called AutoML Translation which still utilizes some of the operators/hooks (not all of them).
I'll create a separate PR to delete operators and hooks that couldn't be utilized by this API.

**Review by eladkal — APPROVED** [review]

**Comment by eladkal** [comment]
tests are failing due to merging https://github.com/apache/airflow/pull/38504
you will need to add commit to add the needed files to the ignore yaml

**Comment by eladkal** [comment]
cc @potiuk I am not sure how we can workaround the error build?

**Comment by molcay** [comment]
Hello @shahar1, Thank you for this PR and [the previous PR](https://github.com/apache/airflow/pull/38418). I tried to use the revert button for [the previous PR](https://github.com/apache/airflow/pull/38418) but it did not work.
Actually, I was about to create a PR to make necessary changes and am working on another fork (here is [the PR](https://github.com/VladaZakharova/airflow/pull/35)), but when I created the PR it said that there are conflicts. So I put [the comment](https://github.com/apache/airflow/pull/38418#issuecomment-2025027673) to [the previous PR](https://github.com/apache/airflow/pull/38418).

When I create the community PR (to this repo), I will ask a review from you @shahar1

**Comment by shahar1** [comment]
> Hello @shahar1, Thank you for this PR and [the previous PR](https://github.com/apache/airflow/pull/38418). I tried to use the revert button for [the previous PR](https://github.com/apache/airflow/pull/38418) but it did not work.
> Actually, I was about to create a PR to make necessary changes and am working on another fork (here is [the PR](https://github.com/VladaZakharova/airflow/pull/35)), but when I created the PR it said that there are conflicts. So I put [the comment](https://github.com/apache/airflow/pull/38418#issuecomment-2025027673) to [the previous PR](https://github.com/apache/airflow/pull/38418).
> 
> When I create the community PR (to this repo), I will ask a review from you @shahar1

Hey, there was a conflict due to changes of dependcies handling in `pyproject.toml`.
I've already created PR #38635 for the deprecation here - I didn't know that you've already created this PR, as you created it in Vlada's fork and not here :)
After merging the current PR - you may recreate the PR here, I'll close mine and review yours - or we do the other way around, let me know whay works best for you.

**Comment by molcay** [comment]
Hey @shahar1,


Thank you again for your effort and struggles for this matter; very appreciated. 

---
> ...I didn't know that you've already created this PR, as you created it in Vlada's fork and not here :)

Yes, I know I created [the PR ](https://github.com/VladaZakharova/airflow/pull/35)in Vlada's fork 😅 It was my fault for not mentioning the WIP PR. Sorry for the miscommunication.
Vlada's fork is our first place to create PR and get internal review then we are creating the PR to this repo. When I was creating the PR in the Vlada's fork, I saw that there were (and still are, _since this is not merged yet_) some conflicts I put that comment to #38418. But couldn't revert (not before you at least 😅)

Btw, I created the new PR: https://github.com/apache/airflow/pull/38673

--- 
> After merging the current PR - you may recreate the PR here, I'll close mine and review yours ...

If we can do this, it will be wonderful because for [the PR](https://github.com/apache/airflow/pull/38673) I tried to comply with the [Airflow Deprecation Policy](https://airflow.apache.org/docs/apache-airflow/stable/release-process.html#deprecation-policy).
> **e.g:** Giving alternative operators instead of the deprecated and/or shutdown services' operators

Also, we will continue to support most of the operators (at least for the next major version of the package) due to the deprecation policy. They will definitely fail and we will raise an exception for the users who are using the operators but it needs to be there. Hence we cannot remove the system tests.

Also, the main idea of the new PR is finding alternatives/replacements of the operators for the deprecated services instead of just removing them.

---

To summarize the current situation,
We have 4 PRs:
| PR     | Summary                                                   |
|--------|-----------------------------------------------------------|
| #38418 | Merged and need to be reverted                            |
| #38633 | This PR reverting #38418                                  |
| #38635 | The other PR related to this topic                        |
| #38673 | This is the new PR that contains policy complaint changes |

**Review by potiuk — APPROVED** [review]
